### PR TITLE
Add incident_editor and admin roles to submission edit

### LINKED ISSUE #2606: Edit Submission Form – `incident_editor` seems unable to accept or reject incident reports
Yesterday, I changed @datherton09's AIID account from `admin` to `incident_editor` and `taxonomy_editor` as part of:
* https://github.com/responsible-ai-collaborative/aiid/issues/2383#issue-1968598857

 However, he then reported being unable to accept or reject incident report submissions, with the corresponding UI being disabled/greyed out:

image

I reverted his account back to `admin` permissions and this resolved the issue. This leads me to believe there is a codepath in the editing flow still requiring `admin` permission.

### PR DESCRIPTION
**Pull request by clari182**
- closes #2606 

Users with `incident_editor` should now be able to review and approve/reject a submission on the submitted page

Test URL: https://pr-43--clarayoudale-staging.netlify.app/apps/submitted/
Ping Clara for user/password login information for an incident_editor role user
Otherwise you can also test on your own env with a custom user.

### CODE REVIEW

**Review by pdcp1 — CHANGES_REQUESTED** [review]
I tried to Reject this submission https://pr-43--clarayoudale-staging.netlify.app/apps/submitted/?editSubmission=65c2a1b7fa1e108114b37fa3 logged as clara_editor@botsfactory.io and I got this error in the console:

```
Uncaught (in promise) ApolloError: reason="role \"default\" in \"aiidprod.submissions\" does not have delete permission for document with _id: ObjectID(\"65c2a1b7fa1e108114b37fa3\")"; code="ArgumentsNotAllowed"; untrusted="delete not permitted"; details=map[]
```

In addition to that, we should display an error toast message when an error occurs to let the user know that it failed.

image

File: site/realm/data_sources/mongodb-atlas/aiidprod/submissions/rules.json [hunk 1]
```diff
@@ -13,6 +13,17 @@
...
+            "name": "incident_editor",
+            "apply_when": {
+                "%%user.custom_data.roles": "incident_editor"
+            },
+            "read": true,
+            "write": true,
+            "insert": true,
+            "delete": true,
+            "search": true
+        },
```
Comment (pdcp1) [inline site/realm/data_sources/mongodb-atlas/aiidprod/submissions/rules.json:26]: Add this new rule to the wiki page https://github.com/responsible-ai-collaborative/aiid/wiki/DB-schema-rules

**Review by pdcp1 — COMMENTED** [review]
@clari182 Can you whitelist the https://pr-43--clarayoudale-staging.netlify.app on your env so I can test this?

File: site/realm/data_sources/mongodb-atlas/aiidprod/submissions/rules.json (hunk 1 above)
Comment (clari182) [reply site/realm/data_sources/mongodb-atlas/aiidprod/submissions/rules.json:26]: Done, url whitelisted and wiki updated

**Review by pdcp1 — CHANGES_REQUESTED** [review]
@clari182 I get a different error when I try to `Reject` the submission using the clara_editor@botsfactory.io account on your environment https://pr-43--clarayoudale-staging.netlify.app

```
index.js:28 Uncaught (in promise) ApolloError: reason="role \"default\" in \"aiidprod.submissions\" does not have delete permission for document with _id: ObjectID(\"65cccff85934b769475352e7\")"; code="ArgumentsNotAllowed"; untrusted="delete not permitted"; details=map[]
```

In addition to that, I got another error when I tried to promote a submission to a new Incident:

```
index.js:28 Uncaught (in promise) ApolloError: reason="role \"incident_editor\" in \"aiidprod.incidents\" does not have insert permission"; code="ArgumentsNotAllowed"; untrusted="insert not permitted"; details=map[]
```

If the `incident_editor` role should promote/reject submission, it should have `insert` permission to `incident` collection.
Please check the `New issue` feature as well. I guess we have the same problem with `reports` collection, but I'm not sure if it is needed.

**Review by pdcp1 — APPROVED** [review]
LGTM 👍 
Thanks @clari182 for updating the README as well

**Pull request merged**